ndc-reference/bin/reference/main.rs

```diff
@@ -1878,10 +1908,47 @@ fn eval_nested_field(
                     }),
                 )
             })?;
-            let result_array = array
-                .into_iter()
+
+            let limit = if apply_array_arguments {
```

Comment:
There is a bug here: if `fields` is not provided, we never reach this case, so we don't apply the limit. We need to lift this logic out into `eval_field`.